src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -52,20 +62,30 @@ class UDPPeerGroup[F[_]](val udpPeerGroupConfig: Config)(implicit liftF: Lift[F]
   }
 
   private def writeUdp(address: InetSocketAddress, data: ByteBuffer): Unit = {
-    val udp = DatagramChannel.open()
-    udp.configureBlocking(true)
-    udp.connect(address)
     try {
-      udp.write(data)
-    } finally {
-      udp.close()
+      val udp = DatagramChannel.open()
+      udp.configureBlocking(true)
```

Comment:
Can't this fail? if it can, this should be inside the try.

Reply:
We could put it in a try but, really, if it does fail, it's a bug in the library:-)